google/cloud/spanner_v1/_pandas_helpers.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2021 Google LLC
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     https://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import warnings
+
+SPANNER_TO_PANDAS_DTYPES = {
+    "INT64": "int64",
+    "STRING": "object",
+    "BOOL": "bool",
+    "BYTES": "object",
+    "ARRAY": "object",
+    "DATE": "datetime64[ns]",
+    "FLOAT64": "float64",
+    "NUMERIC": "object",
+    "TIMESTAMP": "datetime64[ns]",
+}
+
+
+def pd_dataframe(response_obj):
+    """This functions converts the query results into pandas dataframe
+
+    :rtype: pandas.Dataframe
+    :returns: Dataframe with the help of a mapping dictionary which maps every spanner datatype to a pandas compatible datatype.
+    """
+    try:
+        from pandas import DataFrame
+    except ImportError as err:
+        raise ImportError(
+            "Pandas module not found. It is needed for converting query result to Dataframe.\n Try running 'pip3 install pandas'"
+        ) from err
+
+    data = []
+    for row in response_obj:
+        data.append(row)
```

Comment:
I believe this can be simplified to:

```suggestion
    data = list(response_obj)
```